main.js

```diff
@@ -10,3 +31,51 @@ function connectToDatabase(dbName) {
     };
     return level(dbName, options);
 }
+
+async function getNext(iterator) {
+    return new Promise((resolve, reject) => {
+        iterator.next((err, key, value) => {
+            if (err) {
+                reject(err);
+            } else if (key && value) {
+                resolve({ key: key, value: value });
+            } else {
+                iterator.end(() => {
+                    resolve(undefined);
+                })
+            }
+        });
+    })
+}
+async function listAllStudents(db) {
+    let iterator = db.iterator();
+    let nextKeyValue;
+    do {
+        nextKeyValue = await getNext(iterator);
+        if (nextKeyValue) {
+            const student = nextKeyValue.value;
+            console.log('\nID No.: ',student.ID,'\nFullname: ', student.name,'\nAge: ', student.age, '\nAddress: ', student.address,    '\nStatus: ', student.status);
+        }
+    } while (nextKeyValue);
+}
+
+async function acceptStudent(db, ID, studentName, studentAge, studentAddress) {
+    const studentInfo = {
+        ID: ID,
+        name: studentName,
+        age: studentAge,
+        address: studentAddress,
+        status: 'Applying'
+    };
+    return db.put(ID, studentInfo);
+}
+async function scheduleInterview(db, ID, ScheduleDate) {
+    try {
+        const student =  await db.get(ID);
+        student.interviewSched = ScheduleDate;
+        student.status = 'Under Interview';
+        db.put(ID, student);
```

Comment:
you need to use await on db.put to make sure changes takes effect before proceeding
```suggestion
        db.put(ID, student);
```